src/fixture.ts

```diff
@@ -30,30 +36,62 @@ const removeTrailingPytestInfo = (lines: string[]) => {
  * @param output pytest --fixtures output
  * @returns list of fixtures parsed from output
  */
-const parsePytestOutputToFixtures = (output: string) => {
+const parsePytestOutputToFixtures = (output: string, rootDir: string) => {
     const fixtures: Fixture[] = [];
     let lines = removeTrailingPytestInfo(output.split("\n"));
+    let alreadyEncountered: Record<string, number> = {};
+    let tmpContent: string[] | null = null;
+    let currentFilePath: string|null = null;
+
     let fixture: Fixture = {
         name: "",
         docstring: "",
     };
+
+    function append(fixture: Fixture) {
+        let index = alreadyEncountered[fixture.name] ?? -1;
+        if(index >= 0) {
+            fixtures[index] = fixture;
+        }
+        else {
+            fixtures.push(fixture);
+            alreadyEncountered[fixture.name] = fixtures.length - 1;
+        }
+    }
+
     lines.forEach(line => {
+        let matches;
+
         // Two spaces means docstring or error, pytest includes no docstring errors if there are no docstrings
         if (line.startsWith("  ") && !line.includes("no docstring")) {
             fixture.docstring += `\n${line}`;
-        } else if (line.match(/^[\w]/i)) { // If the line starts with a letter or a number, we assume fixture
+        } else if (matches = line.match(/^(\w+) -- ([^:]+):(\d+)$/i)) { // If the line starts with a letter or a number, we assume fixture
             if (fixture.name) {
-                fixtures.push(fixture);
+                append(fixture);
+            }
+            const [name, linePath, line] = matches.slice(1);
+            fixture = { name, docstring: "" };
+
+            if(linePath !== currentFilePath)
+            {
+                currentFilePath = linePath;
+                tmpContent = readFileSync(join(rootDir, linePath), "utf8").split(EOL);
             }
-            fixture = {
-                name: line.split(" ")[0].trim(),
-                docstring: "",
-            };
+
+            fixture.fileLocation = vscode.Uri.file(join(rootDir, linePath));
+            const lineInt = parseInt(line, 10) - 1;
+            const start = tmpContent![lineInt].indexOf(`def ${fixture.name}(`) + 4;
+            const end = start + fixture.name.length;
+            fixture.range = new vscode.Range(
+                new vscode.Position(lineInt, start),
+                new vscode.Position(lineInt, end)
+            );
```

Comment:
suggestion: guard this with `if (tmpContent && tmpContent.length > lineInt)` in case something goes wrong with parsing